core/src/main/java/org/apache/mesos/scheduler/plan/DefaultBlockFactory.java

```diff
@@ -50,6 +50,10 @@ public Block getBlock(String taskType, TaskSpecification taskSpecification) thro
             } catch (InvalidTaskSpecificationException e) {
                 logger.error("Failed to generate TaskSpecification for existing Task with exception: ", e);
                 throw new InvalidRequirementException(e);
+            } catch (InvalidProtocolBufferException e) {
+                logger.error("Failed to unpack taskInfo: {}", e);
+                throw new InvalidRequirementException(
+                        "Failed to generate correct TaskSpecification for existing Task with exception: ", e);
```

Comment:
This error message is identical to the catch clause above.  It should be updated to match the logged error.